BlockSettleUILib/PubKeyLoader.cpp

```diff
@@ -42,100 +44,29 @@ BinaryData PubKeyLoader::loadKey(const KeyType kt) const
-#ifndef PRODUCTION_BUILD
-   case ApplicationSettings::EnvConfiguration::Staging:
-      filename += QStringLiteral("staging");
-      break;
-   case ApplicationSettings::EnvConfiguration::Custom:
-      filename += QStringLiteral("prod");
-      break;
-#endif
-   }
-   if (filename.isEmpty()) {
-      return {};
-   }
-   else {
-      filename = QStringLiteral(":/resources/PublicKeys/") + filename
-         + QStringLiteral(".key");
-   }
-   QFile f(filename);
-   if (!f.open(QIODevice::ReadOnly)) {
-      return {};
-   }
-   BinaryData result;
-   try {
-      result.createFromHex(f.readAll().toStdString());
-   }
-   catch (...) {
-      return {};
-   }
-   return result;
+   return {};
 }
 
-bool PubKeyLoader::saveKey(const KeyType kt, const BinaryData &key)
-{
-   switch (kt) {
-   case KeyType::Chat:
-      appSettings_->set(ApplicationSettings::chatServerPubKey, QString::fromStdString(key.toHexStr()));
-      break;
-   case KeyType::Proxy:
-      appSettings_->set(ApplicationSettings::proxyServerPubKey, QString::fromStdString(key.toHexStr()));
-      break;
-   case KeyType::CcServer:
-      appSettings_->set(ApplicationSettings::ccServerPubKey, QString::fromStdString(key.toHexStr()));
-      break;
-   case KeyType::ExtConnector:
-      appSettings_->set(ApplicationSettings::ExtConnPubKey, QString::fromStdString(key.toHexStr()));
-      break;
-   }
-   return true;
-}
 
 bs::network::BIP15xNewKeyCb PubKeyLoader::getApprovingCallback(const KeyType kt
-   , QWidget *bsMainWindow, const std::shared_ptr<ApplicationSettings> &appSettings)
+   , QWidget *bsMainWindow, const std::shared_ptr<ApplicationSettings> &appSettings
+   , const std::shared_ptr<BootstrapDataManager>& bootstrapDataManager)
 {
    // Define the callback that will be used to determine if the signer's BIP
    // 150 identity key, if it has changed, will be accepted. It needs strings
    // for the old and new keys, and a promise to set once the user decides.
    //
    // NB: This may need to be altered later. The PuB key should be hard-coded
    // and respected.
-   return [kt, bsMainWindow, appSettings] (const std::string& oldKey
+   return [kt, bsMainWindow, appSettings, bootstrapDataManager] (const std::string& oldKey
          , const std::string& newKeyHex, const std::string& srvAddrPort
          , const std::shared_ptr<FutureValue<bool>> &newKeyProm) {
-      const auto &deferredDialog = [kt, bsMainWindow, appSettings, newKeyHex, newKeyProm, srvAddrPort]{
-         QMetaObject::invokeMethod(bsMainWindow, [kt, bsMainWindow, appSettings, newKeyHex, newKeyProm, srvAddrPort] {
-            PubKeyLoader loader(appSettings);
+      const auto &deferredDialog = [kt, bsMainWindow, appSettings, bootstrapDataManager, newKeyHex, newKeyProm, srvAddrPort]{
+         QMetaObject::invokeMethod(bsMainWindow, [kt, bsMainWindow, appSettings, bootstrapDataManager, newKeyHex, newKeyProm, srvAddrPort] {
+            PubKeyLoader loader(appSettings, bootstrapDataManager);
```

Comment:
The new key prompt GUI should warn the user the old key (local one) is signed and the new key (from remote) is not

Reply:
Updated